proposals/lip-0045.md

```diff
@@ -984,164 +839,223 @@ The Interoperability module contains the following commands:
+    return createTerminatedStateAccount(chainID)
+```
 
-The `getChainAccount` function returns the object stored in the chain data substore with store key equal to the input `chainID`.
+#### error
 
-##### Parameters
+The `error` function is used to add an error code to a CCM and then add it to the outbox of a partner chain.
 
-* `chainID`: The ID of the chain.
+```python
+def error(ccm: CCM, errorStatus: uint32, relatedCCMIDs: list[bytes]) -> None:
+    # Error codes from 0 to MAX_RESERVED_ERROR_STATUS (included) are reserved to the Interoperability module
+    if 0 <= errorStatus <= MAX_RESERVED_ERROR_STATUS:
+        raise Exception('Invalid error status.')
+    
+    originalCCMID = SHA-256(encode(crossChainMessageSchema, ccm))
+    return send(
+        EMPTY_FEE_ADDRESS,
+        ccm.moduleID,
+        ccm.crossChainCommandID,
+        ccm.sendingChainID,
+        0,
+        errorStatus,
+        ccm.params
+    )
+```
 
-##### Returns
+#### getChainAccount
 
-This function returns `chainAccount(chainID)`.
+The `getChainAccount` function returns the object stored in the chain data substore with store key equal to the input `chainID: ChainID`.
+It returns `chainAccount(chainID)`.
 
 #### getChannel
 
-The `getChannel` function returns the object stored in the channel data substore with store key equal to the input `chainID`.
-
-##### Parameters
-
-* `chainID`: The ID of the chain.
-
-##### Returns
-
-This function returns `channel(chainID)`.
+The `getChannel` function returns the object stored in the channel data substore with store key equal to the input `chainID: ChainID`.
+It returns `channel(chainID)`.
 
 
 #### getOwnChainAccount
 
 The `getOwnChainAccount` function returns the object stored in the own chain data substore.
-
-##### Parameters
-
-This function takes no input.
-
-##### Returns
-
-This function returns `ownChainAccount`.
+It returns `ownChainAccount`.
 
 #### getTerminatedStateAccount
 
-The `getTerminatedStateAccount` function returns the object stored in the terminated state substore with store key equal to the input `chainID`.
-
-##### Parameters
-
-* `chainID`: The ID of the chain.
-
-##### Returns
-
-This function returns `terminatedStateAccount(chainID)`.
+The `getTerminatedStateAccount` function returns the object stored in the terminated state substore with store key equal to the input `chainID: ChainID`.
+It returns `terminatedStateAccount(chainID)`.
 
 #### getTerminatedOutboxAccount
 
-The `getTerminatedOutboxAccount` function returns the object stored in the terminated outbox substore with store key equal to the input `chainID`.
-
-##### Parameters
-
-* `chainID`: The ID of the chain.
-
-##### Returns
-
-This function returns `terminatedOutboxAccount(chainID)`.
+The `getTerminatedOutboxAccount` function returns the object stored in the terminated outbox substore with store key equal to the input `chainID: ChainID`.
+It returns `terminatedOutboxAccount(chainID)`.
 
 #### isDirectChannelActive
```

Comment:
I am not sure about the purpose of this method. For chains that exists, the status can be "registered", "active" or "terminated". This method would return "active" also for chains for which the function `isLive` would return `False`, which is kind of misleading. Wouldn't it be more useful to get the chain status instead?

Reply:
I think it will be used by the direct sidechain LIP. We could also remove it for the moment and only add it later.

Reply:
I would remove this function then. It is also an API (not an internal function).

Reply:
removed